ote_sdk/ote_sdk/utils/argument_checks.py

```diff
@@ -0,0 +1,217 @@
+    file_extension = file_path.split(".")[-1].lower()
+    if file_extension not in expected_extensions:
+        raise ValueError(
+            f"Unexpected extension of {file_path_name} file. expected: {expected_extensions} actual: {file_extension}"
+        )
+
+
+def check_that_null_character_absents_in_string(parameter: str, parameter_name: str):
+    """Function raises ValueError exception if null character: '\0' is specified in string"""
+    if "\0" in parameter:
+        raise ValueError(f"\\0 is specified in {parameter_name}: {parameter}")
+
+
+def check_that_file_exists(file_path: str, file_path_name: str):
+    """Function raises ValueError exception if file not exists"""
+    if not exists(file_path):
+        raise ValueError(
+            f"File {file_path} specified in '{file_path_name}' parameter not exists"
+        )
+
+
+def check_file_path(file_path: str, file_path_name: str, expected_extensions: list):
+    """
+    Function raises ValueError exception if non-string object is specified as file path, if file has unexpected
+    extension or if file not exists
+    """
+    raise_value_error_if_parameter_has_unexpected_type(
+        parameter=file_path, parameter_name=file_path_name, expected_type=str
+    )
+    check_that_string_not_empty(string=file_path, parameter_name=file_path_name)
+    check_file_extension(
+        file_path=file_path,
+        file_path_name=file_path_name,
+        expected_extensions=expected_extensions,
+    )
+    check_that_null_character_absents_in_string(
+        parameter=file_path, parameter_name=file_path_name
+    )
+    check_that_file_exists(file_path=file_path, file_path_name=file_path_name)
+
+
+def check_input_config_parameter(input_config):
+    """
+    Function raises ValueError exception if "input_config" parameter is not equal to expected
+    """
+    parameter_name = "input_config"
+    raise_value_error_if_parameter_has_unexpected_type(
+        parameter=input_config,
+        parameter_name=parameter_name,
+        expected_type=(str, DictConfig, dict),
+    )
+    if isinstance(input_config, str):
+        check_that_string_not_empty(string=input_config, parameter_name=parameter_name)
+        check_that_null_character_absents_in_string(
+            parameter=input_config, parameter_name=parameter_name
+        )
+        if isinstance(safe_load(input_config), str):
+            check_file_extension(
+                file_path=input_config,
+                file_path_name=parameter_name,
+                expected_extensions=["yaml"],
+            )
+            check_that_file_exists(
+                file_path=input_config, file_path_name=parameter_name
+            )
+    if isinstance(input_config, dict):
+        if input_config == {}:
+            raise ValueError(
+                "Empty dictionary is specified as 'input_config' parameter"
+            )
+
+
+def check_is_parameter_like_dataset(parameter, parameter_name):
```

Comment:
Please, see the comment above on `BaseInputArgumentChecker` and path checks.
I think it could be better to make special `class DatasetParameterChecker(BaseInputArgumentChecker)` and use it.